### ISSUE TITLE
Extract tool issues

### ISSUE THREAD

**Issue by Sfshaza** [flutter/flutter#7889]
There are a couple problems with the extract tool (tool/extract.dart) on the flutter website.

* It doesn't recurse. When it was implemented, the flutter website had a flat structure, but I'm adding some subdirectories. Those aren't being checked.

* It's too strict. It's fine to check the code snippets for errors and warnings, but checking for hints is too draconian. It's not unusual to have code snippets with an unused variable, for example, and even that causes a failure.

**Comment by Hixie** [comment]
To avoid the hints, there's a trick where you put a comment before the snippet that has code in it that declares all the implicit variables you're using.

**Comment by Hixie** [comment]
(You do want to have all the hints because otherwise when the framework changes, it'll just change from having no warnings to having ignored warnings, instead of catching the problem.)

**Comment by Sfshaza** [comment]
Thanks, @hixie. I'll have to experiment with that comment. I definitely don't want to ignore warnings or errors.

**Comment by sethladd** [comment]
We've addressed #1 with putting samples into their own directories, and our travis now analyzes those files.

We can control #2 with analyzer options files and line-level ignores. I'd like to open a new issue specific for "how to best analyze our snippets for our website". We may want to check for some hints, but not all.

Closing this for now, but when you're ready to do line-level ignores, we'll help you out. Thanks!
